sos/plugins/gnocchi.py

```diff
@@ -70,18 +56,64 @@ def setup(self):
                 "gnocchi resource-type list"
             ])
 
-    def postproc(self):
-        self.do_file_sub(
-            "/etc/gnocchi/gnocchi.conf",
-            r"password=(.*)",
-            r"password=*****",
+    def apply_regex_sub(self, regexp, subst):
+        self.do_path_regex_sub("/etc/gnocchi/*", regexp, subst)
+        self.do_path_regex_sub(
+            self.var_puppet_gen + "/etc/gnocchi/*",
+            regexp, subst
         )
-        self.do_file_sub(
-            self.var_puppet_gen + "/etc/gnocchi/"
-            "gnocchi.conf",
-            r"password=(.*)",
-            r"password=*****",
+
+    def postproc(self):
+        protect_keys = ["ceph_secret", "password", ]
+        self.apply_regex_sub(
+            r"((?m)^\s*(%s)\s*=\s*)(.*)" % "|".join(protect_keys),
+            r"\1*********"
```

Comment:
This seems a bit awkward to me. I'd much rather we stay consistent with other plugin's where the `postproc()` call handles all the substitutions. Here I don't see the benefit to making a new method when all we're doing is consolidating the constructed regex for the same files.